pkg/monitor/agent.go

```diff
@@ -29,55 +30,65 @@ import (
 type agentMonitor struct {
 	client  clientset.Interface
 	querier agentquerier.AgentQuerier
+	// agentCRD is the desired state of agent monitoring CRD which agentMonitor expects.
+	agentCRD *v1beta1.AntreaAgentInfo
 }
 
 // NewAgentMonitor creates a new agent monitor.
 func NewAgentMonitor(client clientset.Interface, querier agentquerier.AgentQuerier) *agentMonitor {
-	return &agentMonitor{client: client, querier: querier}
+	return &agentMonitor{client: client, querier: querier, agentCRD: nil}
 }
 
 // Run creates AntreaAgentInfo CRD first after controller is running.
 // Then updates AntreaAgentInfo CRD every 60 seconds.
 func (monitor *agentMonitor) Run(stopCh <-chan struct{}) {
 	klog.Info("Starting Antrea Agent Monitor")
-	agentCRD := monitor.getAgentCRD()
-	var err error = nil
 
-	// Initialize agent monitoring CRD.
-	if agentCRD == nil {
-		agentCRD, err = monitor.createAgentCRD()
-		if err != nil {
-			klog.Errorf("Failed to create agent monitoring CRD %+v: %v", agentCRD, err)
-			return
-		}
-	} else {
-		agentCRD, err = monitor.updateAgentCRD(agentCRD)
-		if err != nil {
-			klog.Errorf("Failed to update agent monitoring CRD %+v: %v", agentCRD, err)
+	// Sync agent monitoring CRD every minute util stopCh is closed.
+	wait.Until(monitor.syncAgentCRD, time.Minute, stopCh)
+}
+
+func (monitor *agentMonitor) syncAgentCRD() {
+	var err error = nil
+	if monitor.agentCRD != nil {
+		if monitor.agentCRD, err = monitor.updateAgentCRD(true); err == nil {
 			return
+		} else {
+			klog.Errorf("Failed to partially update agent monitoring CRD: %v", err)
```

Comment:
nit: Since the above condition has `return`, no need to have indent for this path.